demos/template/templates/modules/post.html

```diff
@@ -0,0 +1,9 @@
+{% extends "{ROOT_DIR}/templates/common.html" %}
```

Comment:
>  If we use relative paths, this will result in a path error.

Why? There must be *some* relative path that would work. Do you just mean that it's messy to count the right number of `../`s to use?

If you want multiple roots, it seems to me that it would be more natural to use a search path (as in `$PATH` or `sys.path`) instead of explicit variable substitution.

Reply:
```
TEMPLATE_DIR
	├─base
	│  └─base.html
	├─common
	│  └─common.html		# some common code
	├─normal
	│  └─normal.html
	└─some_extension
		└─extension
		   ├─extension_one.html
		   └─extension_two.html
```

	   
In this case,if we want normal.html and extension_one.html extends or include common.html,and common.html extends base.html with relative path,either normal.html or extension_one.html will get a path error.because directory hierarchies is different.

So,if we defined a variable TEMPLATE_DIR,wo can use a path like {TEMPLATE_DIR}/base/base.html in common.html to solve this problem.
Or define directory base to a variable BASE_DIR,and use a path like {BASE_DIR}/base.html in common.html

Reply:
It sounds like a search path would solve the same problem, then. I think I'd rather have that than a new variable substitution method.